dao/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.exception.WrongInsertDataException;
+import com.epam.esm.domain.entity.Tag;
+import com.epam.esm.dao.mapper.TagMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Component
+public class TagDaoImpl implements TagDao {
+
+    private static final String SELECT_TAG_BY_NAME = "SELECT * FROM tag WHERE name=?";

```

Comment:
names of columns will be exactly better than *

Reply:
i'll remember. fixed